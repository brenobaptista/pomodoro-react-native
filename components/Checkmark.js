import * as React from 'react';
import { View, StyleSheet } from 'react-native';
import Feather from 'react-native-vector-icons/Feather';
import { connect } from 'react-redux';

class Checkmark extends React.Component {
  render() {
    return (
      <View>
        <View style={styles.row}>
          <Feather
            name={this.props.check1 ? 'check-square' : 'square'}
            size={35}
            color="#4a4a4a"
          />
          <Feather
            name={this.props.check2 ? 'check-square' : 'square'}
            size={35}
            color="#4a4a4a"
          />
          <Feather
            name={this.props.check3 ? 'check-square' : 'square'}
            size={35}
            color="#4a4a4a"
          />
          <Feather
            name={this.props.check4 ? 'check-square' : 'square'}
            size={35}
            color="#4a4a4a"
          />
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  row: {
    flexDirection: 'row',
    justifyContent: 'space-evenly',
  },
});

const mapStateToProps = state => {
  return {
    check1: state.check.checkmark1,
    check2: state.check.checkmark2,
    check3: state.check.checkmark3,
    check4: state.check.checkmark4,
  }
};

export default connect(mapStateToProps)(Checkmark);
