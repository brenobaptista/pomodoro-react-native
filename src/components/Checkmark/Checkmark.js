import React from 'react';
import { View } from 'react-native';
import Feather from 'react-native-vector-icons/Feather';

const Checkmark = (props) => (
  <View>
    <Feather
      name={props.check ? 'check-square' : 'square'}
      size={35}
      color="#4a4a4a"
    />
  </View>
);

export default Checkmark;
