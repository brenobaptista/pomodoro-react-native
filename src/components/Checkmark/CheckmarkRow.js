import React from 'react';
import { View, StyleSheet } from 'react-native';
import { connect } from 'react-redux';
import Checkmark from './Checkmark';

const CheckmarkRow = (props) => {
  return (
    <View>
      <View style={styles.row}>
        <Checkmark check={props.check1} />
        <Checkmark check={props.check2} />
        <Checkmark check={props.check3} />
        <Checkmark check={props.check4} />
      </View>
    </View>
  )
};

const styles = StyleSheet.create({
  row: {
    flexDirection: 'row',
    justifyContent: 'space-evenly',
  },
});

const mapStateToProps = state => {
  return {
    check1: state.check.checkmark1,
    check2: state.check.checkmark2,
    check3: state.check.checkmark3,
    check4: state.check.checkmark4,
  };
};

export default connect(mapStateToProps)(CheckmarkRow);
