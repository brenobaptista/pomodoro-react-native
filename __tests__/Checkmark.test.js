import React from 'react';
import { shallow } from 'enzyme';
import Checkmark from '../src/components/Checkmark/Checkmark';
import toJson from 'enzyme-to-json';

describe('Checkmark', () => {
    describe('Rendering', () => {
        it('should match to snapshot - True', () => {
            const component = shallow(<Checkmark check={true} />)
            expect(toJson(component)).toMatchSnapshot("True checkmark snapshot")
        });
        it('should match to snapshot - False', () => {
            const component = shallow(<Checkmark check={false}/>)
            expect(toJson(component)).toMatchSnapshot("False checkmark snapshot")
        });
    });
});