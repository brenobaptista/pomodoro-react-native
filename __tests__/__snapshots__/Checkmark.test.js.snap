// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkmark Rendering should match to snapshot - False: False checkmark snapshot 1`] = `
<View>
  <Icon
    allowFontScaling={false}
    color="#4a4a4a"
    name="square"
    size={35}
  />
</View>
`;

exports[`Checkmark Rendering should match to snapshot - True: True checkmark snapshot 1`] = `
<View>
  <Icon
    allowFontScaling={false}
    color="#4a4a4a"
    name="check-square"
    size={35}
  />
</View>
`;
